### DISCUSSION TITLE
Can i use socket.io-client to connect to a POS thermal printer

### DISCUSSION THREAD

**Discussion by lakshmajee** [socketio/socket.io#3799] [Q&A] [answered]
Can i use `socket.io-client` library to connect to POS thermal printer, which supports TCP/IP?

**Comment by darrachequesne** [comment, marked as answer]
The Socket.IO client is meant to be run either in the browser, or with Node.js. So I'd say no (unless your printer can actually run Node.js, of course).

**Reply by lakshmajee** [reply]
Agreed, my printer don't understand NodeJS :smile:

**Reply by lakshmajee** [reply]
And I was able to connect to printer thru socket connections from an android application (code was written in java). No drivers / external dependencies were installed.
